# Correlation Analysis

__In order to find the interdependency of the variables DistanceFromHome, MonthlyIncome,
TotalWorkingYears, YearsAtCompany, YearsWithCurrManager from that of Attrition, we executed
the Correlation Analysis as follows.__

In [1]:
import pandas as pd

In [2]:
dataset = pd.read_csv("C:/Users/RAHUL REDDY/Desktop/AIML/Letsupgrade/Assignment/general_data.csv")
dataset

Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
1      31       Yes  Travel_Frequently  Research & Development   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely                   Sales   
4409   40        No      Travel_Rarely  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   
4409                28          3        Medical              1        4410   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0     Female  ...                 1.0      Y                11              8   
1     Female  ...                 0.0      Y                23              8   
2       Male  ...                 1.0      Y                15              8   
3       Male  ...                 3.0      Y                11              8   
4       Male  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4405  Female  ...                 3.0      Y                17              8   
4406    Male  ...                 2.0      Y                15              8   
4407    Male  ...                 0.0      Y                20              8   
4408    Male  ...                 0.0      Y                14              8   
4409    Male  ...                 0.0      Y                12              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   
4409                 0               NaN                      6   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...                   ..

In [13]:
from sklearn import preprocessing
l = preprocessing.LabelEncoder()
attrition = l.fit_transform(dataset["Attrition"])

In [14]:
attrition

array([0, 1, 0, ..., 0, 0, 0])

In [10]:
from scipy.stats import pearsonr
dataset['TotalWorkingYears']=dataset['TotalWorkingYears'].fillna(11.28)
dataset.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'EmployeeCount', 'EmployeeID', 'Gender',
       'JobLevel', 'JobRole', 'MaritalStatus', 'MonthlyIncome',
       'NumCompaniesWorked', 'Over18', 'PercentSalaryHike', 'StandardHours',
       'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear',
       'YearsAtCompany', 'YearsSinceLastPromotion', 'YearsWithCurrManager'],
      dtype='object')

In [15]:
stats, p = pearsonr(attrition, dataset.DistanceFromHome)
print(stats, p)

-0.009730141010179674 0.5182860428050771


In [16]:
stats, p=pearsonr(attrition, dataset.MonthlyIncome)
print(stats, p)

-0.031176281698115007 0.03842748490600132


In [17]:
stats, p=pearsonr(attrition, dataset.TotalWorkingYears)
print(stats, p)

-0.17011136355965176 5.473159751791005e-30


In [18]:
stats, p=pearsonr(attrition, dataset.YearsAtCompany)
print(stats, p)

-0.1343922139899772 3.1638831224877484e-19


In [20]:
stats, p=pearsonr(attrition , dataset.YearsWithCurrManager)
print(stats, p)


-0.15619931590162847 1.7339322652896276e-25


___The inference of the above analysis are as follows:__  
  
__Attrition & DistanceFromHome:__  
As r = -0.009, there’s low negative correlation between Attrition and DistanceFromHome  
As the P value of 0.518 is > 0.05, we are accepting H0 and hence there’s no significant correlation
between Attrition & DistanceFromHome  
  
__Attrition & MonthlyIncome:__  
As r = -0.031, there’s low negative correlation between Attrition and MonthlyIncome  
As the P value of 0.038 is < 0.05, we are accepting Ha and hence there’s significant correlation
between Attrition & MonthlyIncome  
  
__Attrition & TotalWorkingYears:__  
As r = -0.17, there’s low negative correlation between Attrition and TotalWorkingYears  
As the P value is < 0.05, we are accepting Ha and hence there’s significant correlation between
Attrition & TotalWorkingYears  
  
__Attrition & YearsAtCompany:__  
As r = -0.1343, there’s low negative correlation between Attrition and YearsAtCompany  
As the P value is < 0.05, we are accepting Ha and hence there’s significant correlation between
Attrition & YearsAtCompany  
  
__Attrition & YearsWithCurrManager:__  
As r = -0.1561, there’s low negative correlation between Attrition and YearsWithCurrManager  
As the P value is < 0.05, we are accepting Ha and hence there’s significant correlation between
Attrition & YearsWithCurrManager.